lib/mixpanel-ruby/consumer.rb

```diff
@@ -88,7 +94,11 @@ def send(type, message)
       form_data = {"data" => data, "verbose" => 1}
       form_data.merge!("api_key" => api_key) if api_key
 
-      response_code, response_body = request(endpoint, form_data)
+      begin
+        response_code, response_body = request(endpoint, form_data)
+      rescue Exception => e
+        raise ConnectionError.new("Could not connect to Mixpanel, with error \"#{e.message}\".")
```

Comment:
This loses the stack trace and meta-data of e (which is better than throwing Net::CrazyUnpredictableException) - we might want to include ConnectionError#requestError or something. (I believe that this might automagically be populated as ConnectionError#cause in Ruby 2.1, which might be worth verifying)

This also might be over-general? (We can't really tell if request throws a MyAwesomeHTTPLibException, and we want to catch that, but on the other hand this will also catch a YourComputerIsOnFireException, which we probably want to let through...) - This is a trade off, but it's a trade off worth thinking about?